source/IntegrationTests/Behaviours/IncomingRequests/GivenAggregatedMeasureDataRequestTests.cs

```diff
@@ -177,4 +232,129 @@ public async Task
             .HasReceiverId("2111111111111")
             .HasOriginalTransactionIdReference("123564789123564789123564789123564787");
     }
+
+    /// <summary>
+    /// Even though an actor has delegated his requests to another actor, he should still
+    /// be able to request and receive his own data
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(DocumentFormatsWithActorRoleCombinations))]
+    public async Task AndGiven_DataInTwoGridAreas_When_ActorPeeksAllMessages_Then_ReceivesTwoNotifyAggregatedMeasureDataDocumentWithCorrectContent(ActorRole actorRole, DocumentFormat incomingDocumentFormat, DocumentFormat peekDocumentFormat)
+    {
+        /*
+         *  --- PART 1: Receive request, create process and send message to Wholesale ---
+         */
+
+        // Arrange
+        var senderSpy = CreateServiceBusSenderSpy();
+        var currentActor = (ActorNumber: ActorNumber.Create("1111111111111"), ActorRole: actorRole);
+        var energySupplierNumber = currentActor.ActorRole == ActorRole.EnergySupplier
+            ? currentActor.ActorNumber
+            : ActorNumber.Create("3333333333333");
+        var balanceResponsibleParty = currentActor.ActorRole == ActorRole.BalanceResponsibleParty
+            ? currentActor.ActorNumber
+            : ActorNumber.Create("4444444444444");
+
+        GivenNowIs(Instant.FromUtc(2024, 7, 1, 14, 57, 09));
+        GivenAuthenticatedActorIs(currentActor.ActorNumber, currentActor.ActorRole);
+
+        // Original actor requests own data
+        await GivenReceivedAggregatedMeasureDataRequest(
+            documentFormat: incomingDocumentFormat,
+            senderActorNumber: currentActor.ActorNumber,
+            senderActorRole: currentActor.ActorRole,
+            meteringPointType: MeteringPointType.Consumption,
```

Comment:
```suggestion
            meteringPointType: MeteringPointType.Consumption,
```
is it possible to get one, where meteringPointType = null ?

Reply:
I think that makes a lot of sense. I've considered adding behaviour tests for request aggregated time series and request wholesale services without delegation, and i think i'll do that after this PR is merged